stock_request_purchase/models/purchase_order.py

```diff
@@ -22,6 +23,52 @@ def _compute_stock_request_ids(self):
             rec.stock_request_ids = rec.order_line.mapped("stock_request_ids")
             rec.stock_request_count = len(rec.stock_request_ids)
 
+    def button_cancel(self):
+        super().button_cancel()
+        self._exception_on_stock_request()
+
+    def _get_exception_on_stock_request_info(self):
```

Comment:
Why do you need so much code for this easy one? And adding an extra method is not needed, as this is not called from several places. Do you need any hook?

Reply:
This method is needed to cover all use cases:
- Purchase linked to stock requests.
- Purchase linked to a child production linked to a stock request (https://github.com/OCA/stock-logistics-warehouse/pull/1837) **Important**: In this case it will be necessary to call the method before cancellation to continue to have the linked productions.

What do you consider would be the correct way to proceed in this case?

Reply:
That "inherited link" is not something of this module, and the propagation of the cancellation shouldn't be treated at that level. For me this should only handle the direct link.